x/mongo/driver/topology/connection_options.go

```diff
@@ -70,7 +70,7 @@ func newConnectionConfig(opts ...ConnectionOption) (*connectionConfig, error) {
 	}
 
 	if cfg.dialer == nil {
-		cfg.dialer = &net.Dialer{Timeout: cfg.connectTimeout}
+		cfg.dialer = &net.Dialer{}
```

Comment:
Why are we removing the default  30 second `Timeout` for the dialer here?

Reply:
Previously, setting the `Timeout` field made it so `connectTimeout` would be applied to the `DialContext` call to make a socket. Now, we're using `context.WithTimeout(ctx, connectTimeout)` and passing the context down to the `DialContext` and `Handshake` calls to apply it to both socket establishment and TLS handshaking, so setting `Timeout` is no longer required.

Reply:
Gotcha. That sounds good!